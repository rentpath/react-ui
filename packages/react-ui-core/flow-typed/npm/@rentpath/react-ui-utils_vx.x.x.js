// flow-typed signature: a137d1bccb9d876def54c58cc5b6e8c1
// flow-typed version: <<STUB>>/@rentpath/react-ui-utils_v0.0.1/flow_v0.49.1

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@rentpath/react-ui-utils'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@rentpath/react-ui-utils' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@rentpath/react-ui-utils/lib/composeProps' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/lib/index' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/lib/keyMirror' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/lib/parseArgs' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/lib/randomId' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/__tests__/composeProps-test' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/__tests__/keyMirror-test' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/__tests__/parseArgs-test' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/__tests__/randomId-test' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/composeProps' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/index' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/keyMirror' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/parseArgs' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/src/randomId' {
  declare module.exports: any;
}

declare module '@rentpath/react-ui-utils/test/setup' {
  declare module.exports: any;
}

// Filename aliases
declare module '@rentpath/react-ui-utils/lib/composeProps.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/composeProps'>;
}
declare module '@rentpath/react-ui-utils/lib/index.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/index'>;
}
declare module '@rentpath/react-ui-utils/lib/keyMirror.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/keyMirror'>;
}
declare module '@rentpath/react-ui-utils/lib/parseArgs.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/parseArgs'>;
}
declare module '@rentpath/react-ui-utils/lib/randomId.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/lib/randomId'>;
}
declare module '@rentpath/react-ui-utils/src/__tests__/composeProps-test.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/__tests__/composeProps-test'>;
}
declare module '@rentpath/react-ui-utils/src/__tests__/keyMirror-test.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/__tests__/keyMirror-test'>;
}
declare module '@rentpath/react-ui-utils/src/__tests__/parseArgs-test.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/__tests__/parseArgs-test'>;
}
declare module '@rentpath/react-ui-utils/src/__tests__/randomId-test.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/__tests__/randomId-test'>;
}
declare module '@rentpath/react-ui-utils/src/composeProps.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/composeProps'>;
}
declare module '@rentpath/react-ui-utils/src/index.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/index'>;
}
declare module '@rentpath/react-ui-utils/src/keyMirror.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/keyMirror'>;
}
declare module '@rentpath/react-ui-utils/src/parseArgs.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/parseArgs'>;
}
declare module '@rentpath/react-ui-utils/src/randomId.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/src/randomId'>;
}
declare module '@rentpath/react-ui-utils/test/setup.js' {
  declare module.exports: $Exports<'@rentpath/react-ui-utils/test/setup'>;
}
