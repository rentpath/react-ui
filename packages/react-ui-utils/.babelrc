{
  "env": {
    "es": {
      "plugins": [
        "transform-decorators-legacy",
        "transform-class-properties",
        "babel-plugin-transform-object-rest-spread",
      ]
    },
    "cjs": {
      "presets": [
        "env"
      ],
      "plugins": [
        "transform-runtime",
        "transform-decorators-legacy",
        "transform-class-properties",
        "babel-plugin-transform-object-rest-spread"
      ]
    },
    "test": {
      "presets":[
        "env",
        "react"
      ],
      "plugins": [
        "transform-runtime",
        "transform-decorators-legacy",
        "transform-class-properties",
        "babel-plugin-transform-object-rest-spread"
      ]
    }
  }
}
