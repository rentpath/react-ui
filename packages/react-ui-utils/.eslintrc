{
  "parser": "babel-eslint",
  "extends": "airbnb-base",
  "rules": {
    "semi": [2, "never"],
    "arrow-parens": [2, "as-needed"],
    "no-underscore-dangle": [0],
    "import/no-extraneous-dependencies": ["error", {
      "devDependencies": true,
      "packageDir": "./package.json"
    }]
  },
  "settings": {
    "import/resolver": {
      "node": {
        "moduleDirectory": [
          "node_modules",
          "packages"
        ]
      }
    }
  },
  "env": {
    "mocha": true,
    "browser": true,
    "node": true
  }
}
