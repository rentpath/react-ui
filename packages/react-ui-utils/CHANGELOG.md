# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="2.0.0"></a>
# [2.0.0](https://github.com/rentpath/react-ui/compare/@rentpath/react-ui-utils@1.0.3...@rentpath/react-ui-utils@2.0.0) (2018-04-03)


### Features

* **dependencies:** upgrade packages ([ca579bd](https://github.com/rentpath/react-ui/commit/ca579bd))


### BREAKING CHANGES

* **dependencies:** Now requires react ^16.3.0 as a peer




<a name="1.0.3"></a>
## [1.0.3](https://github.com/rentpath/react-ui/compare/@rentpath/react-ui-utils@1.0.2...@rentpath/react-ui-utils@1.0.3) (2018-01-31)


### Bug Fixes

* **babel:** updated babel preset to env from es2015. ([efdde4d](https://github.com/rentpath/react-ui/commit/efdde4d))




<a name="1.0.2"></a>
## [1.0.2](https://github.com/rentpath/react-ui/compare/@rentpath/react-ui-utils@1.0.0...@rentpath/react-ui-utils@1.0.2) (2018-01-17)




**Note:** Version bump only for package @rentpath/react-ui-utils
