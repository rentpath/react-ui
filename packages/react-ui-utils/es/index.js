export { default as randomId } from './randomId';
export { default as parseArgs } from './parseArgs';
export { default as composeProps } from './composeProps';
export { default as keyMirror } from './keyMirror';
export { default as componentName } from './componentName';