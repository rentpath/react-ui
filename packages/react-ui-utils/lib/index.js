'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _randomId = require('./randomId');

Object.defineProperty(exports, 'randomId', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_randomId).default;
  }
});

var _parseArgs = require('./parseArgs');

Object.defineProperty(exports, 'parseArgs', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_parseArgs).default;
  }
});

var _composeProps = require('./composeProps');

Object.defineProperty(exports, 'composeProps', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_composeProps).default;
  }
});

var _keyMirror = require('./keyMirror');

Object.defineProperty(exports, 'keyMirror', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_keyMirror).default;
  }
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }