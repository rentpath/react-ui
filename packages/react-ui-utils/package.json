{
  "name": "@rentpath/react-ui-utils",
  "version": "3.0.1",
  "main": "lib/index.js",
  "license": "MIT",
  "scripts": {
    "clean": "rm -rf lib",
    "build": "babel --ignore __tests__ -d lib src",
    "build:watch": "yarn run build -- --watch",
    "test": "NODE_ENV=test jest",
    "test:watch": "yarn run test -- --watch",
    "lint": "eslint src",
    "lint:fix": "yarn run lint -- --fix"
  },
  "devDependencies": {
    "@rentpath/eslint-config-rentpath": "3.0.1",
    "babel-cli": "6.26.0",
    "babel-core": "6.26.0",
    "babel-eslint": "^8.2.2",
    "babel-jest": "^22.4.3",
    "babel-plugin-transform-object-rest-spread": "6.26.0",
    "babel-preset-env": "1.6.1",
    "eslint": "^4.19.1",
    "jest": "^22.4.3"
  },
  "dependencies": {
    "classnames": "2.2.5"
  }
}
