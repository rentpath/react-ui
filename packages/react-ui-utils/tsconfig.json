{
  "compilerOptions": {
    "module": "commonjs",
    "lib": ["es2015", "dom", "es2017.object"],
    "outDir": "./lib",
    "target": "es2015",
    "allowJs": true,
    "removeComments": true,
    "pretty": true
  },
  "include": [
    "src"
  ],
  "exclude": [
    "**/__tests__/*"
  ]
}