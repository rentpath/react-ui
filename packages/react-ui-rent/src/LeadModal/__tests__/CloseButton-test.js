import React from 'react'
import renderer from 'react-test-renderer'
import { keyMirror } from '@rentpath/react-ui-utils'
import ThemedCloseButton from '../CloseButton'

const CloseButton = ThemedCloseButton.WrappedComponent

const theme = keyMirror([
  'Text',
  'TermsOfService',
])

describe('CloseButton', () => {
  it('renders default children if none passed', () => {
    const snap = renderer
      .create(<CloseButton theme={theme} />)
      .toJSON()
    expect(snap).toMatchSnapshot()
  })

  it('renders children passed', () => {
    const snap = renderer
      .create(<CloseButton theme={theme}>Close</CloseButton>)
      .toJSON()

    expect(snap).toMatchSnapshot()
  })
})
