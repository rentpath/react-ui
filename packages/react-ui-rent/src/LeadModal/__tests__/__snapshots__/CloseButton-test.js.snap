// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton renders children passed 1`] = `
<div
  className=""
>
  Close
</div>
`;

exports[`CloseButton renders default children if none passed 1`] = `
<div
  className=""
>
  X
</div>
`;
